### PR TITLE
Add Rest API

### PR DESCRIPTION
**Pull request by NyCodeGHG**

### CODE REVIEW

**Comment by NyCodeGHG** [comment]
cloudflare why are you deploying this tf

**Comment by DRSchlaubi** [comment]
> cloudflare why are you deploying this tf

They just deploy everything

**Comment by NyCodeGHG** [comment]
They deploy themselfes 😱

**Review by warriorzz — APPROVED** [review]

File: rest-api/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  mavenCentral()
+  maven("https://jitpack.io")
+}
+
+dependencies {
+  implementation("io.ktor", "ktor-server-core", "1.5.3")
+  implementation("io.ktor", "ktor-server-netty", "1.5.3")
+  implementation("io.ktor", "ktor-serialization", "1.5.3")
+  implementation("io.ktor", "ktor-locations", "1.5.3")
+  testImplementation("io.ktor", "ktor-server-test-host", "1.5.3")
```
Comment (DRSchlaubi) [inline rest-api/build.gradle.kts:17]:
```suggestion
  implementation(platform("io.ktor:ktor-bom:1.5.3")
  implementation("io.ktor", "ktor-server-core")
  implementation("io.ktor", "ktor-server-netty")
  implementation("io.ktor", "ktor-serialization")
  implementation("io.ktor", "ktor-locations")
  testImplementation("io.ktor", "ktor-server-test-host")
```

File: rest-api/src/main/kotlin/v1/ApiVersionOne.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@OptIn(KtorExperimentalLocationsAPI::class)
+fun Route.apiV1() = route("v1") {
+  val ageService: AgeService by inject()
+  get<RootRoute> {
+    if (it.format !in arrayOf("message", "period")) {
+      throw InvalidApiFormatException(it.format)
+    }
+
+    when (it.format) {
+      "message" -> {
```
Comment (DRSchlaubi) [inline rest-api/src/main/kotlin/v1/ApiVersionOne.kt:25]: Might enum this. DK if locations supports that

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: rest-api/src/main/kotlin/v1/ApiVersionOne.kt (hunk 2 above)
Comment (NyCodeGHG) [reply rest-api/src/main/kotlin/v1/ApiVersionOne.kt]: It's supported with [Data Conversion](https://ktor.io/docs/data-conversion.html)

**Comment by DRSchlaubi** [comment]
AND WHEN I GOOO SURF THE INTERWEBS
MY BROWSER IS CHROME

**Comment by NyCodeGHG** [comment]
BUT LINUS SAID IT WAS OKAY

**Comment by NyCodeGHG** [comment]
@DRSchlaubi you have done nothing for the last half hour except talk to me on teamspeak, you could have easily looked at my PR so please do that now, thank you

**Comment by DRSchlaubi** [comment]
After some evaluation, I decided that we don't need such a feature yet, so I'll close this for now

**Pull request closed without merging**